src/mem/sizeclass.h

```diff
@@ -174,4 +174,32 @@ namespace snmalloc
     return relative;
   }
 #endif
+
+  inline static size_t aligned_size(size_t alignment, size_t size)
```

Comment:
Could you add an `SNMALLOC_FAST_PATH` here.  The codegen is currently putting a function call on the fast path.

Ultimately I think for good codegen, we will need something like, 
```C++
is_aligned_size(alignment, size)
```
which returns true, if the size meets the alignment requirements.  Or build a table lookup so this can be much fast.